pydantic/dataclasses.py

```diff
@@ -185,7 +185,7 @@ def create_dataclass(cls: type[Any]) -> type[PydanticDataclass]:
 
         cls = dataclasses.dataclass(  # type: ignore[call-overload]
             cls,
-            init=init,
+            init=True,
```

Comment:
Should we add a note about this here as well?

Reply:
I added a comment just above it